voltixApp/VoltixApp/Services/BalanceService.swift

```diff
@@ -8,76 +8,60 @@
 import Foundation
 
 class BalanceService {
-
+    
     static let shared = BalanceService()
-
+    
     private let utxo = BlockchairService.shared
     private let thor = ThorchainService.shared
-    private let eth = EtherScanService.shared
-    private let bsc = BSCService.shared
-    private let avax = AvalancheService.shared
     private let sol = SolanaService.shared
     private let gaia = GaiaService.shared
-
+    
     func balance(for coin: Coin) async throws -> (coinBalance: String, balanceUSD: String) {
         await CryptoPriceService.shared.fetchCryptoPrices()
-
-        switch coin.chain {
-        case .bitcoin, .bitcoinCash, .litecoin, .dogecoin:
+        
+        switch coin.chainType {
+        case .UTXO:
```

Comment:
I prefer to use switch on `coin.chain` , it is much better than using `chainType`  , chainType is a really weird property , for example , bitcoin is UTXO chain , that's fine , but .THORChain is not a chain type , Solana is not a chain type either  
Could you please revert  it to use `coin.chain`?   and also since `coin.chain` is enum , so if the case is not exhaustive , it will give us an error

Reply:
Done